loom/common/core/allocator.h

```diff
@@ -22,6 +22,9 @@
 #define _CORE_ALLOCATOR_H_
 
 #include "loom/common/core/assert.h"
+#include "jemalloc/jemalloc.h"
+#include <stdint.h>
```

Comment:
Not 100% on cross-platformness, but they're used for `bool` and e.g. `uint32_t`. I should move them to `allocator.c` anyway, since header file doesn't use them anymore. In any case, if we run into trouble we can include LuaJIT's compat version.